api/middleware/authRequired.js

```diff
@@ -19,23 +19,32 @@ const makeProfileObj = (claims) => {
  * if the token is not present or fails validation. If the token is valid its
  * contents are attached to req.profile
  */
+
 const authRequired = async (req, res, next) => {
   try {
     const authHeader = req.headers.authorization || '';
     const match = authHeader.match(/Bearer (.+)/);
-
     if (!match) throw new Error('Missing idToken');
 
     const idToken = match[1];
     oktaJwtVerifier
       .verifyAccessToken(idToken, oktaVerifierConfig.expectedAudience)
       .then(async (data) => {
         const jwtUserObj = makeProfileObj(data.claims);
-        const profile = await Model.findOrCreate('seller_profile', jwtUserObj);
-        if (profile) {
-          req.profile = profile;
+        const sellerProfile = await Model.findOrCreate(
+          'seller_profile',
+          jwtUserObj
+        );
+        const buyerProfile = await Model.findOrCreate(
+          'buyer_profile',
+          jwtUserObj
+        );
+        if (sellerProfile) {
+          req.profile = sellerProfile;
+        } else if (buyerProfile) {
+          req.profile = sellerProfile;
         } else {
-          throw new Error('Unable to process idToken');
+          throw new Error('Undable to process idToken');
```

Comment:
small typo here. everything else looks good though.

Reply:
Good catch just submitted another commit fixing the typo